### ISSUE TITLE
Can't seem to get it to work...

### ISSUE THREAD

**Issue by JelmerOT** [cpsoinos/nuxt-svgo#223]
I can't get my icon to show up... this is my nuxt.config.ts:
```javascript
['nuxt-svgo', {
  defaultImport: 'component',
  customComponent: 'Icon',
  svgoConfig: {
    multipass: true,
    plugins: [{
      name: 'preset-default',
      params: {
        overrides: {
          removeViewBox: false,
        },
      },
    }],
  },
}],
```

My custom icon.vue component:
```vue
<template>
  <Component
    :is="icon"
    :class="{
      'fill-current': filled,
      'h-[1em] w-[1em]': fontControlled,
    }"
  />
</template>

<script setup lang="ts">
withDefaults(defineProps<{
  icon: string;
  filled?: boolean;
  fontControlled?: boolean;
}>(), {
  filled: true,
  fontControlled: false,
});
</script>
```

And my icon SVG is placed in assets/icons/plus.svg. SVG looks like this:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path d="M16 32c-.68 0-1.23-.55-1.23-1.23V17.23H1.23a1.23 1.23 0 1 1 0-2.46h13.54V1.23a1.23 1.23 0 1 1 2.46 0v13.54h13.54a1.23 1.23 0 1 1 0 2.46H17.23v13.54c0 .68-.55 1.23-1.23 1.23Z" /></svg>
```

Usage in a component:
```vue
<Icon icon="plus" class="inline-block h-10 w-10" />
```

Any help is appreciated :-)

**Comment by jd1378** [comment]
Hi
Please read the readme again for his

**Comment by renardsas** [comment]
Hi @JelmerOT 
A delayed response, but if others encounter the same issue as you, I’ll detail what @jd1378  means when he says to reread the documentation.

There are 3 errors in your code:
**1. Nuxt config**
_defaultImport: 'component'_: As mentioned in the documentation, by doing this, you're indicating that you don't want to use nuxt-icon and here don't want to use your custom component that replaces nuxt-icon : Icon.
So, you shouldn't include this.

See : [https://github.com/cpsoinos/nuxt-svgo?tab=readme-ov-file#usage](https://github.com/cpsoinos/nuxt-svgo?tab=readme-ov-file#usage)

**2. Your custom component**
The icon prop is an object , not a string.
See : [https://github.com/cpsoinos/nuxt-svgo/blob/main/src/runtime/components/nuxt-icon.vue](https://github.com/cpsoinos/nuxt-svgo/blob/main/src/runtime/components/nuxt-icon.vue)

**3. Your usage of the custom component**
`
<Icon icon="plus" class="inline-block h-10 w-10" />
`
You shouldn't call it like that. Instead, as indicated in the documentation: Your custom component is called Icon, so:
`
<IconPlus class="inline-block h-10 w-10" />
`

See : [https://github.com/cpsoinos/nuxt-svgo?tab=readme-ov-file#nuxt-icon-component](https://github.com/cpsoinos/nuxt-svgo?tab=readme-ov-file#nuxt-icon-component)

So to summarize, here’s the corrected code:

```js
// nuxt.config.ts
['nuxt-svgo', {
  customComponent: 'Icon',
  svgoConfig: {
    multipass: true,
    plugins: [{
      name: 'preset-default',
      params: {
        overrides: {
          removeViewBox: false,
        },
      },
    }],
  },
}],

// component Icon
<template>
  <Component
    :is="icon"
    :class="{
      'fill-current': filled,
      'h-[1em] w-[1em]': fontControlled,
    }"
  />
</template>

<script setup lang="ts">
withDefaults(defineProps<{
  icon: object;
  filled?: boolean;
  name?: string // Add optional prop name.
  fontControlled?: boolean;
}>(), {
  filled: true,
  fontControlled: false,
});
</script>

// Usage
<IconPlus class="inline-block h-10 w-10" />
```
